File: js/vector_to_array.js
const puerts = require("puerts");

let vector_to_array = puerts.load('../vector_to_array/vector_to_array');

const VectorTest = vector_to_array.VectorTest

const input = [0, 1, 2, 3, 4, 5, 0, 1, 2, 3, 4, 5 ];
const output = VectorTest.convert(input);

console.log(output);

File: vector_to_array/src/vector_to_array.cc
#include <functional>
#include <string>
#include <iostream>
#include <vector>
#include "Binding.hpp"
#include "vector_to_array.h"

class VectorTest
{
public:
    static std::vector<int> convert(std::vector<int> input) {
        for (int i = 0; i < input.size(); i++) {
            input[i]++;
        }
        return input;
    }
};


// module declaration begin

UsingCppType(VectorTest);

void Init() {
    puerts::DefineClass<VectorTest>()
        .Function("convert", MakeFunction(&VectorTest::convert))
        .Register();
}

PESAPI_MODULE(vector_to_array, Init)

// module declaration end
